tests/bluetooth/bsim_bt/bsim_test_mesh/src/distribute_keyid.h

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef TESTS_BLUETOOTH_BSIM_BT_BSIM_TEST_MESH_SRC_DISTRIBUTE_KEYID_H_
+#define TESTS_BLUETOOTH_BSIM_BT_BSIM_TEST_MESH_SRC_DISTRIBUTE_KEYID_H_
+
+#if defined CONFIG_BT_MESH_USES_MBEDTLS_PSA
+void stored_keys_clear(void);
+#else
+static inline void stored_keys_clear(void)
```

Comment:
This is header. What I said is the truth for *.c files. If remove inlining here it'll cause syntax error.

Reply:
It will cause a situation when you get several function bodies with the same name via reincluding header. :)
Finally, it causes a compiler or linker error (do not remember which one will generate it).
For header functions `static inline` is mandatory or at least wishible to avoid this. For source file functions it doesn't have sense.

Reply:
> It will cause a situation when you get several function bodies with the same name via reincluding header. :)

No, it won't as long as header is guarded.

> Finally, it causes a compiler or linker error (do not remember which one will generate it).

No, this is not how static keyword works.

Reply:
> > It will cause a situation when you get several function bodies with the same name via reincluding header. :)
> 
> No, it won't as long as header is guarded.
> 

Some headers are guarded some aren't. Common and good practice to not put function bodies in the headers other than static inline functions. It protects from what I described above. I do not see any reason to do oppositely to worldwide practice.  

> > Finally, it causes a compiler or linker error (do not remember which one will generate it).
> 
> No, this is not how static keyword works.

Hm. My sentences are correct in the pair. Both describe exact compiler\linker behavior. In general, it doesn't matter function is static or isn't. Without guarding the issue will be. Static keyword will not help. No sense to discuss the second part without the first.

Reply:
Your header is guarded. So no issue for this case. But generally I agree if you define 2 functions with the same name in one unit, this will cause linker error. But it is not the case for this.